### PR TITLE
cmd/atlas/internal: rough deployment implementation

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

**Review by giautm — APPROVED** [review]

File: cmd/atlas/internal/cmdapi/migrate.go [hunk 1]
```diff
@@ -229,41 +244,160 @@ func migrateApplyRun(cmd *cobra.Command, args []string, flags migrateApplyFlags)
...
+// RecordTargetID asks the revisions-table to allow or provide
+// the target identifier if cloud reporting is enabled.
+func (r *MigrateReport) RecordTargetID(ctx context.Context) error {
+	if r.CloudEnabled() {
+		id, err := r.rrw.ID(ctx, operatorVersion())
 		if err != nil {
-			err2 = fmt.Errorf("%w: %v", err, err2)
+			return err
 		}
-		err = err2
```
Comment (masseelch) [inline cmd/atlas/internal/cmdapi/migrate.go:261]: <3

File: cmd/atlas/internal/cmdapi/migrate.go [hunk 2]
```diff
@@ -1645,9 +1779,8 @@ func isURL(s string) bool {
 
 // cmdEnvsRun executes a given command on each of the configured environment.
 func cmdEnvsRun[F any](
-	run func(*cobra.Command, []string, F) error,
-	set func(*cobra.Command, *Env) error,
-	cmd *cobra.Command, args []string, flags *F,
+	flags *F, setFlags func(*cobra.Command, *Env) error,
+	cmd *cobra.Command, runCmd func(*Env) error,
```
Comment (masseelch) [inline cmd/atlas/internal/cmdapi/migrate.go:1783]:
```suggestion
	flags *F, 
	setFlags func(*cobra.Command, *Env) error,
	cmd *cobra.Command, 
	runCmd func(*Env) error,
```
Prefer this for readability

**Review by masseelch — APPROVED** [review]
:rocket: Can't wait to see this in action :-)

File: cmd/atlas/internal/cmdapi/migrate.go (hunk 2 above)
Comment (a8m) [reply cmd/atlas/internal/cmdapi/migrate.go]: Fixed

**Pull request merged**